mm2src/coins/nft.rs

```diff
@@ -465,26 +736,46 @@ fn check_moralis_ipfs_bafy(token_uri: Option<&str>) -> Option<String> {
     })
 }
 
-async fn get_uri_meta(token_uri: Option<&str>, metadata: Option<&str>) -> UriMeta {
+async fn get_uri_meta(token_uri: Option<&str>, metadata: Option<&str>, url_antispam: &Url) -> UriMeta {
     let mut uri_meta = UriMeta::default();
+    // Fetching data from the URL if token_uri is provided
     if let Some(token_uri) = token_uri {
-        if let Ok(response_meta) = send_request_to_uri(token_uri).await {
-            if let Ok(token_uri_meta) = serde_json::from_value(response_meta) {
-                uri_meta = token_uri_meta;
-            }
+        if let Some(url) = construct_camo_url_with_token(token_uri, url_antispam) {
```

Comment:
I am interested to know how camo will be used for `image_url`/`animation_url`/etc.. , will the url be constructed from the GUI side? or will it not be used at all?

Reply:
@smk did `/url/decode/{url_hex}` endpoint capable to get json object or if link returns image type content you will download picture, if request was successful of course(camo doesnt allow redirections). So yeah, GUI also should use camo to download pictures

Reply:
> So yeah, GUI also should use camo to download pictures

Should we provide them the constructed camo url or will they construct it from their side? It can be an option in `NftMetadataReq`, if requested, we would then convert the urls after we retrieve them from database to camo format.

Reply:
> Should we provide them the constructed camo url or will they construct it from their side?

GUI needs to convert `image_url` string to hex format and send get req to [`/url/decode/{url_hex}`](https://nft.antispam.dragonhound.info/docs#/Encode%20%2F%20Decode%20Urls/Hex_Url_Proxy_url_decode__url_hex__get). There is no need to build the whole complex camo url, proxy does it on its side. All you need is to provide hex format. I think GUI can do it on their side.

> It can be an option in `NftMetadataReq`, if requested, we would then convert the urls after we retrieve them from database to camo format.

As far as I know they send get nft list req with `max` flag to see all nft info and then try to send req to `image_url` to download picture for all items. But of course we can speak about it and if they want we can add ready hex format.